test/models/drive_test.dart

```diff
@@ -0,0 +1,1003 @@
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.pending,
+            seats: 1),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.pending,
+            seats: 3),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.cancelledByRider,
+            seats: 2),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.rejected,
+            seats: 2),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.withdrawnByRider,
+            seats: 1),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.cancelledByDriver,
+            seats: 2),
+      ]);
+      expect(drive.getMaxUsedSeats(), 0);
+    });
+    test('returns 1 if rides has only 1 approved ride with 1 seat', () async {
+      final Drive drive = DriveFactory().generateFake(seats: 3, rides: <Ride>[
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.approved,
+            seats: 1),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.pending,
+            seats: 3),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.cancelledByRider,
+            seats: 2),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.rejected,
+            seats: 2),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.withdrawnByRider,
+            seats: 1),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.cancelledByDriver,
+            seats: 2),
+      ]);
+      expect(drive.getMaxUsedSeats(), 1);
+    });
+    test('returns 2 if rides has 2 approved ride with 1 seat at the same time', () async {
+      final Drive drive = DriveFactory().generateFake(seats: 3, rides: <Ride>[
+        RideFactory().generateFake(
```

Comment:
This, too. You want to establish that for rides at the same time, seats will be summed up. So I'd just create two rides at the same time, maybe one at a different time for good measure (even though that would arguably be a different test). Remove the rest, please. As a maintainer, this would be very time-consuming to read in order to understand the relatively basic message scenario this test conveys/tests